cpp/src/arrow/util/tracing_internal.h

```diff
@@ -88,9 +88,10 @@ Iterator<T> WrapIterator(
 
 template <typename T>
 AsyncGenerator<T> WrapAsyncGenerator(AsyncGenerator<T> wrapped,
+                                     opentelemetry::trace::StartSpanOptions options,
                                      const std::string& span_name) {
```

Comment:
Nit:  It's not obvious to me at a glance that I'd be bypassing the normal process of assigning a parent by using this overload. I'm not certain we want to do this but if the overload was...

```
AsyncGenerator<T> WrapAsyncGenerator(AsyncGenerator<T> wrapped, const std::string& span_name, nostd::variant<SpanContext, opentelemetry::context::Context> parent = SpanContext::GetInvalid())
```

It would be more obvious to me:

1. Why someone would supply options
2. By supplying options, I am bypassing the normal process of assigning the parent

Do we call this overload anywhere?  If not, can we get rid of it?

Reply:
duplicate